AsyncDisplayKit/Layout/ASLayoutElement.mm

```diff
@@ -254,6 +254,22 @@ - (void)setPreferredSize:(CGSize)preferredSize
   ASLayoutElementStyleCallDelegate(ASLayoutElementStyleHeightProperty);
 }
 
+- (CGSize)preferredSize
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  if (_size.width.unit != ASDimensionUnitPoints) {
+    NSCAssert(NO, @"preferredSize.width needs to be in ASDimensionUnitPoints");
+    return CGSizeZero;
+  }
+  
+  if (_size.height.unit != ASDimensionUnitPoints) {
+    NSCAssert(NO, @"preferredSize.height needs to be in ASDimensionUnitPoints");
+    return CGSizeZero;
```

Comment:
Let's change the copy to make it clear _why_ we had to assert, and what they should do to fix it.

`Cannot get preferredSize of element with fractional height. Element: <ASTextNode: 0xFFFFFFFF>, height: 25%.`

Use the `delegate` property to populate the element string.